sdk/keyvault/azure-security-keyvault-keys/inc/azure/keyvault/keys/list_keys_single_page_result.hpp

```diff
@@ -9,38 +9,90 @@
 
 #pragma once
 
-#include "azure/keyvault/keys/deleted_key.hpp"
-#include "azure/keyvault/keys/json_web_key.hpp"
-#include "azure/keyvault/keys/key_vault_key.hpp"
+#include <azure/core/http/http.hpp>
+#include <azure/core/paged_response.hpp>
 
 #include <azure/keyvault/common/internal/single_page.hpp>
 
-#include <azure/core/http/http.hpp>
+#include "azure/keyvault/keys/deleted_key.hpp"
+#include "azure/keyvault/keys/json_web_key.hpp"
+#include "azure/keyvault/keys/key_vault_key.hpp"
 
+#include <memory>
 #include <vector>
 
 namespace Azure { namespace Security { namespace KeyVault { namespace Keys {
+  class KeyClient;
 
-  struct KeyPropertiesSinglePage : public Azure::Security::KeyVault::_internal::SinglePage
-  {
+  /**
+   * @brief Define a single page to list the keys from the Key Vault.
+   *
+   */
+  class KeyPropertiesSinglePage : public Azure::Core::PagedResponse<KeyPropertiesSinglePage> {
```

Comment:
Should this still be called `SinglePage`?

Reply:
It's just an uggly name for now.  Saving any other name in case we add iterable

Reply:
If we were to add iterable support, if would be on PagedResponse itself. This is the type we want folks to use. So, I recommend using the good name and remove `SinglePage` everywhere.